swarm/api/mocks/resolver.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2017 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package mocks
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type DummyResolver struct{}
+
+func (dummy *DummyResolver) Resolve(name string) (common.Hash, error) {
```

Comment:
DummyResolver is only for tests. Tests currently initialize api like this:

`api := NewApi(dpa, nil)`

nil here corresponding to the dns object.
Now because of this new check we have in if swarm/api/api.go

`self.dns == nil {
￼	return nil, fmt.Errorf("Your node is not connected to the blockchain")
}`

swarm/api/api_test.go and swarm/api/http/server_test.go fail.
In other words, tests were supposed to succeed even if the dns part of the api object was nil.

Maybe there is another way to fix this, but that is what I figured out to be the quickest.